src/inet/ZephyrSocketHacks.h

```diff
@@ -0,0 +1,59 @@
+#/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This header file defines BSD socket API functions which for various
+ *      reasons have not (yet) been implemented or exposed in Zephyr.
+ */
+
+#ifndef INET_ZEPHYR_SOCKET_HACKS_H
+#define INET_ZEPHYR_SOCKET_HACKS_H
+
+#include <sys/socket.h>
+
+static inline int getsockname(int sock, sockaddr * addr, socklen_t * addrlen)
+{
+    return zsock_getsockname(sock, addr, addrlen);
+}
+
+static inline ssize_t sendmsg(int sock, const struct msghdr * msg, int flags)
+{
+    return zsock_sendmsg(sock, msg, flags);
+}
+
+static inline ssize_t recvmsg(int sock, struct msghdr * msg, int flags)
+{
+    // Zephyr doesn't implement recvmsg at all, but if the message vector size is 1 we can simply
+    // translate recvmsg to recvfrom (although we don't get destination interface info in such a case).
+
+    if (msg->msg_iovlen != 1)
```

Comment:
this function would also be ok if it only filled one vector, right?  asking all callers to know that this implementation is limited to a vec of len 1 is brittle.

Reply:
yes, but it's similar to other functions in Zephyr which seem to conform to POSIX/BSD API, but only implement part of their functionality, so such things may happen. Unfortunately, there's no proper recvmsg in Zephyr and although we may work on adding it there, for now we may do what I did or add another set of #ifdefs in Inet layer (which I wanted to avoid). If callers handle errors properly an incorrect use of the function should be spotted quickly.

Reply:
what you're saying (I think) is that we should scrub CHIP for calls to recvmsg() with more than one iov?  in such a case, we should just drop any use of recvmsg().

the alternative (which I propose above) is to handle a multi-iov call by only filling the first iov, which is legal for the definition of recvmsg(), right?

Reply:
I think multi-iov is not a reason why CHIP uses recvmsg(), I may be wrong though. 

But I like the idea to only fill the first iov. I'll provide a new patch once #1594 is merged.

Anyway, you're right that we may only

Reply:
Fixed in the current version